Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -147,102 +189,219 @@ void FieldProbe::ComputeDiags (int step)
-                               xyzmin_arr, amrex::lbound(box), WarpX::n_rz_azimuthal_modes,
-                               interp_order, false);
-
-                // Either save the interpolated fields or the raw fields depending on the raw_fields flag
-                fp_values[index_Ex] = raw_fields ? arrEx(i_probe, j_probe, k_probe) : Ex_interp;
-                fp_values[index_Ey] = raw_fields ? arrEy(i_probe, j_probe, k_probe) : Ey_interp;
-                fp_values[index_Ez] = raw_fields ? arrEz(i_probe, j_probe, k_probe) : Ez_interp;
-                fp_values[index_Bx] = raw_fields ? arrBx(i_probe, j_probe, k_probe) : Bx_interp;
-                fp_values[index_By] = raw_fields ? arrBy(i_probe, j_probe, k_probe) : By_interp;
-                fp_values[index_Bz] = raw_fields ? arrBz(i_probe, j_probe, k_probe) : Bz_interp;
+                // Interpolating to the probe positions for each particle
+
+                // Temporarily defining modes and interp outside ParallelFor to avoid GPU compilation errors.
+                const int temp_modes = WarpX::n_rz_azimuthal_modes;
+                const int temp_interp_order = interp_order;
+                const bool temp_raw_fields = raw_fields;
+                const bool temp_field_probe_integrate = m_field_probe_integrate;
+
+                long const np = pti.numParticles();
+                amrex::ParallelFor( np, [=] AMREX_GPU_DEVICE (long ip)
+                {
+                    amrex::ParticleReal xp, yp, zp;
+                    getPosition(ip, xp, yp, zp);
+
+                    amrex::ParticleReal Exp = 0._rt, Eyp = 0._rt, Ezp = 0._rt;
+                    amrex::ParticleReal Bxp = 0._rt, Byp = 0._rt, Bzp = 0._rt;
+
+                    // first gather E and B to the particle positions
+                    doGatherShapeN(xp, yp, zp, Exp, Eyp, Ezp, Bxp, Byp, Bzp,
+                                   arrEx, arrEy, arrEz, arrBx, arrBy, arrBz,
+                                   Extype, Eytype, Eztype, Bxtype, Bytype, Bztype,
+                                   dx_arr, xyzmin_arr, amrex::lbound(box), temp_modes,
+                                   temp_interp_order, false);
+
+                    //Calculate the Poynting Vector S
+                    amrex::Real const sraw[3]{
+                    Exp * Bzp - Ezp * Byp,
+                    Ezp * Bxp - Exp * Bzp,
+                    Exp * Byp - Eyp * Bxp};
+                    amrex::ParticleReal const S = (1._rt / PhysConst::mu0)  * sqrt(sraw[0] * sraw[0] + sraw[1] * sraw[1] + sraw[2] * sraw[2]);
+
+                    /*
+                     * Determine whether or not to integrate field data.
+                     * If not integrating, store instantaneous values.
+                     */
+                    if (temp_field_probe_integrate)
+                    {
+
+                        // store values on particles
+                        part_Ex[ip] += Exp; //remember to add lorentz transform
```

Comment:
See Dave's comment below:
```suggestion
                        part_Ex[ip] += Exp * dt; //remember to add lorentz transform
```